base/broadcast.jl

```diff
@@ -623,8 +623,44 @@ eltypes(t::Tuple{Any}) = Tuple{_broadcast_getindex_eltype(t[1])}
 eltypes(t::Tuple{Any,Any}) = Tuple{_broadcast_getindex_eltype(t[1]), _broadcast_getindex_eltype(t[2])}
 eltypes(t::Tuple) = Tuple{_broadcast_getindex_eltype(t[1]), eltypes(tail(t)).types...}
 
+function promote_typejoin_union(::Type{T}) where T
+    if T === Union{}
+        return Union{}
+    elseif T isa Union
+        promote_typejoin(promote_typejoin_union(T.a), promote_typejoin_union(T.b))
+    elseif T <: Tuple
+        typejoin_union_tuple(T)
+    else
+        T
+    end
+end
+
+@pure function typejoin_union_tuple(::Type{T}) where {T}
+    p = T.parameters
+    lr = length(p)::Int
+    if lr == 0
+        return Tuple{}
+    end
+    lf, fixed = Core.Compiler.full_va_len(p)
```

Comment:
```suggestion
    lf = lr
```
(`full_va_len` has no applicability here, since it's the number of fields, but you want to iterate over `p`)